modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotChecker.java

```diff
@@ -536,24 +618,145 @@ private boolean hasExpiringEntries(
         return rootIO.getCount(pageAddr) != 0;
     }
 
+    /** Launches local partitions checking and waits for the result, handles execution exceptions. */
+    public Map<PartitionKeyV2, PartitionHashRecordV2> checkPartitionsResult(
+        SnapshotMetadata meta,
+        File snpDir,
+        @Nullable Collection<String> groups,
+        boolean forCreation,
+        boolean checkParts,
+        boolean skipPartsHashes
+    ) {
+        try {
+            return checkPartitions(meta, snpDir, groups, forCreation, checkParts, skipPartsHashes).get();
```

Comment:
Delegate `get()` call to an invoker. Checker should have only one method that checks partitions.